src/session/session_compact.c

```diff
@@ -226,7 +226,7 @@ __compact_checkpoint(WT_SESSION_IMPL *session)
 	 */
 	txn_global = &S2C(session)->txn_global;
 	for (txn_gen = __wt_gen(session, WT_GEN_CHECKPOINT);;) {
-		WT_READ_BARRIER();
+		/* Loop checks volatile-declared objects, no barriers needed. */
```

Comment:
I had to read this comment several times to parse it.   Can we expand the text a bit?  Something like:
* This loop only checks objects that are declared volatile.  Therefore no barriers are needed.